tests/common.go

```diff
@@ -186,6 +186,143 @@ func InitInstance() {
 		err = Inst().Backup.Init(Inst().S.String(), Inst().N.String(), Inst().V.String(), token)
 		expect(err).NotTo(haveOccurred())
 	}
+	if Inst().TestRailMilestoneName != "" {
+		client = testrail.NewClient(Inst().TestRailHostname, Inst().TestRailUsername, Inst().TestRailPassword, true)
+		err := CreateMilestoneIfNotExists()
+		if err != nil {
+			processError(fmt.Errorf("Error creating milestone %s", Inst().TestRailMilestoneName))
+		}
+	} else {
+		fmt.Println("No milestone provided, skipping updates to testrail")
+	}
+}
+
+func CreateMilestoneIfNotExists() error {
+	fmt.Println("Connection to testrail successful")
+	milestones, err := client.GetMilestones(pwxProjectId)
+	if err != nil {
+		fmt.Println("Error in getting milestone:", err)
+		return err
+	}
+	milestoneAlreadyExists := false
+	for _, ms := range milestones {
+		if ms.Name == Inst().TestRailMilestoneName {
+			milestoneAlreadyExists = true
+			milestoneId = ms.ID
+		}
+	}
```

Comment:
I would create a function like GetMilestoneByID which will return either ID or nil. If it is nil then milestone doesn't exist